### PR TITLE
chore(spdk): using spdk-rs crate

### PR DESCRIPTION
**Pull request by dsavitskiy**

### CODE REVIEW

**Comment by mtzaurus** [comment]
bors try

**Comment by dsavitskiy** [comment]
born try

**Comment by dsavitskiy** [comment]
bors try

**Comment by dsavitskiy** [comment]
bors try

**Comment by dsavitskiy** [comment]
bors try

**Comment by dsavitskiy** [comment]
bors try

**Comment by dsavitskiy** [comment]
bors try

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -12,14 +12,14 @@ repos:
     -   id: rust-style
         name: Rust style
         description: Run cargo fmt on files included in the commit. rustfmt should be installed before-hand.
-        entry: cargo fmt --all -- --check
+        entry: cargo +nightly-2021-11-26 fmt --all -- --check
```
Comment (gila) [inline .pre-commit-config.yaml:15]: is the data really requiererd? If so, thats going to be in PITA

**Review by gila — APPROVED** [review]

File: .pre-commit-config.yaml (hunk 1 above)
Comment (mtzaurus) [reply .pre-commit-config.yaml:15]: Something else is also broken:

```
+ nix-shell --run 'cargo fmt --all -- --check' ci.nix

error: attribute '2021-11-26' missing, at /var/lib/jenkins/workspace/Mayastor_trying/nix/lib/rust.nix:7:13

(use '--show-trace' to show detailed location information)

script returned exit code 1
```

**Comment by dsavitskiy** [comment]
bors try

**Comment by dsavitskiy** [comment]
bors try

File: .pre-commit-config.yaml (hunk 1 above)
Comment (dsavitskiy) [reply .pre-commit-config.yaml]: Fixed.

**Review by mtzaurus — APPROVED** [review]

**Pull request merged**